contribution-guide.md

```diff
@@ -0,0 +1,154 @@
+>**Note:** Set the JAVA_HOME environment variable to the path name of the directory into which you installed JDK.
+4. Download [Node.js](https://nodejs.org/en/download/) (version 8.9.x or the latest LTS release) and install it.
+5. Download [npm](https://www.npmjs.com/get-npm) (version 5.6.0 or later) and install it.
+
+## Set up the developer environment
+
+See the below sections to set up IntelliJ IDEA or Eclipse as your development environment.
+
+### Set up IntelliJ IDEA
+
+Follow the steps below to import the Ballerina project to IntelliJ IDEA similar to any other Gradle project.
+
+1. In IntelliJ, click **File** and then click **Open**.
+2. Browse the file system and select the <BALLERINA-LANG_REPO> root directory.
+>**Tip:** If you are opening the directory for the first time in IntelliJ and if prompted to import it as a Gradle project, select the **Auto import** option.
+3. Set Project SDK as Java 1.8.
+4. Install the following plugins:
+    - [Ballerina plugin for IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+    
+        For installation instructions, see [Ballerina IntelliJ Plugin Documentation](https://ballerina.io/learn/tools-ides/intellij-plugin).
+    - [ANTLR plugin for IntelliJ IDEA](https://plugins.jetbrains.com/plugin/7358-antlr-v4-grammar-plugin)
+        
+        This will be useful to check and validate a [grammar rule](#contributing-to-ballerina-grammar) you wrote.
+
+### Set up Eclipse
+
+As all Ballerina repositories are developed as Maven or Gradle projects, follow the steps below to import any Ballerina project to Eclipse similar to any Maven or Gradle project.
+
+1. In Eclipse, click **File** and then click **Import..**.
+2. Under the **Maven** option, select **Existing Maven Projects**. 
+3. Browse the file system and select the Maven module to open it.
+
+## Obtain the source code 
+
+1. Execute the below command to clone the ballerina-lang Git repo.
+```bash 
+git clone --recursive https://github.com/ballerina-platform/ballerina-lang.git
+```
+>**Tip:** If you have already forked the repository to your GitHub account, then execute the below command replacing 'YOUR-GITHUB-USERNAME' with your Git username.
+```bash 
+git clone --recursive https://github.com/<YOUR-GITHUB-USERNAME>/ballerina-lang.git
+```
+2. Execute the below command to update the Git submodules.
+```bash 
+git submodule update --init
+```
+## Build the project
+
+1. Navigate to the <BALLERINA-LANG_REPO> root directory and execute the below Gradle command.
+```bash 
+./gradlew clean build
+```
+2. Extract the built Ballerina distributions in the below locations.
+-  **runtime only:** <BALLERINA-LANG_REPO>/distribution/zip/jballerina/build/distributions/jballerina-<version>-SNAPSHOT.zip. 
+-  **runtime and tools (e.g., Ballerina Composer):** <BALLERINA-LANG_REPO>/distribution/zip/jballerina-tools/build/distributions/jballerina-tools-<version>-SNAPSHOT.zip.
```

Comment:
Since we don't have composer now, we should rephrase this

Reply:
Changed to "runtime and tools (e.g., Ballerina Language Server):".